### PR TITLE
[ Release 2.11 ] upgrade oauth proxy for ocp4.16+

### PR DESCRIPTION
**Pull request by yanmxa**
## Related issue(s)
Reference: https://github.com/stolostron/multiclusterhub-operator/pull/778
Fixes #

### CODE REVIEW

File: operator/pkg/controllers/hubofhubs/globalhub_manager.go [hunk 1]
```diff
@@ -108,11 +108,16 @@ func (r *MulticlusterGlobalHubReconciler) reconcileManager(ctx context.Context,
 		return fmt.Errorf("failed to get the electionConfig %w", err)
 	}
 
+	oauthProxyImage, err := config.GetOauthProxyImage(ctx, r.Manager.GetClient())
```
Comment (clyang82) [inline operator/pkg/controllers/hubofhubs/globalhub_manager.go:111]: why do we need oauth-proxy in global hub manager?

Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/globalhub_manager.go:111]: We use the oauth-proxy for the authentication of the manager non-k8s API.

Comment (clyang82) [reply operator/pkg/controllers/hubofhubs/globalhub_manager.go:111]: I see. Thanks.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**